code/parachain/frame/staking-rewards/src/lib.rs

```diff
@@ -1700,9 +1699,8 @@ pub(crate) fn do_reward_accumulation<T: Config>(
 		let releasable_periods_surpassed =
 			cmp::min(maximum_releasable_periods, periods_surpassed.into());
 
-		let newly_accumulated_rewards = releasable_periods_surpassed
-			.safe_mul(&reward.reward_rate.amount.into())
-			.map_err(|_| RewardAccumulationCalculationError::ArithmeticError)?;
+		let newly_accumulated_rewards =
+			releasable_periods_surpassed.saturating_mul(reward.reward_rate.amount.into());
```

Comment:
If you're going to use saturating, leave a comment about why saturating is safe and doesn't break logic.

Here, just `mul` should be used, and a comment explaining why it's safe